# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: 02cb31bfdc0f
# Full Hash: 02cb31bfdc0f4a15397b3a4e0725c408d738b34c
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Implement VideoDecoder::IsConfigSupported r=padenot
#   
#   This patch implements the isConfigSupported method for WebCodecs'
#   VideoDecoder.
#   
# ==============================================================================

diff -r 77b0f5226ed8 -r 02cb31bfdc0f dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:11 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:11 2023 +0000
@@ -10,6 +10,7 @@
 #include <tuple>
 
 #include "ErrorList.h"
+#include "js/TypeDecls.h"
 #include "mozilla/Result.h"
 #include "mozilla/Span.h"
 #include "mozilla/dom/UnionTypes.h"
@@ -17,6 +18,12 @@
 namespace mozilla::dom {
 
 /*
+ * The followings are helpers for VideoDecoder methods
+ */
+
+nsTArray<nsCString> GuessContainers(const nsString& aCodec);
+
+/*
  * Below are helpers to operate ArrayBuffer or ArrayBufferView.
  */
 
@@ -29,6 +36,11 @@
 Result<Span<uint8_t>, nsresult> GetSharedArrayBufferData(
     const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aBuffer);
 
+Result<OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer, nsresult>
+CloneBuffer(
+    JSContext* aCx,
+    const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aBuffer);
+
 }  // namespace mozilla::dom
 
 #endif  // mozilla_webcodecs_Utils